src/constraints/cost_curve_constraints.jl

```diff
@@ -65,4 +65,144 @@ function add_cost_curve_vars_and_constraints(m, p; _n="")
+            append!(seg_yint_no_inc, 0.0)
+
+            add_to_expression!(m[:CHPCapexNoIncentives],
+                sum(cost_slope_no_inc[s] * m[Symbol("dvSegmentSystemSize"*t)][s] + 
+                    seg_yint_no_inc[s] * m[Symbol("binSegment"*t)][s] for s in 1:p.n_segs_by_tech[t])
+            )
+        else
+            add_to_expression!(m[:CHPCapexNoIncentives], cost_list * m[Symbol("dvPurchaseSize"*_n)]["CHP"])
+        end
+        # TODO: include supplementary firing costs?
+        #Add supplementary firing capital cost
+        # chp_supp_firing_size = self.nested_outputs["Scenario"]["Site"][tech].get("size_supplementary_firing_kw")
+        # chp_supp_firing_cost = self.inputs[tech].get("supplementary_firing_capital_cost_per_kw") or 0
+        # add_to_expression!(m[:CHPCapexNoIncentives], chp_supp_firing_size * chp_supp_firing_cost)
+        add_to_expression!(m[:InitialCapexNoIncentives], m[:CHPCapexNoIncentives])
+    end
+
+    if "SteamTurbine" in p.techs.all
+        add_to_expression!(m[:InitialCapexNoIncentives], 
+            p.s.steam_turbine.installed_cost_per_kw * m[Symbol("dvPurchaseSize"*_n)]["SteamTurbine"]
+        )
+    end
+
+    if "Boiler" in p.techs.all
+        add_to_expression!(m[:InitialCapexNoIncentives], 
+            p.s.boiler.installed_cost_per_kw * m[Symbol("dvPurchaseSize"*_n)]["Boiler"]
+        )
+    end
+
+    if "AbsorptionChiller" in p.techs.all
+        add_to_expression!(m[:InitialCapexNoIncentives], 
+            p.s.absorption_chiller.installed_cost_per_kw * m[Symbol("dvPurchaseSize"*_n)]["AbsorptionChiller"]
+        )
+    end
+
+    if !isempty(p.s.ghp_option_list)
+        for option in enumerate(p.s.ghp_option_list)
+            if option[2].heat_pump_configuration == "WSHP"
+                add_to_expression!(m[:InitialCapexNoIncentives], 
+                    option[2].installed_cost_per_kw[2]*option[2].heatpump_capacity_ton*m[Symbol("binGHP"*_n)][option[1]]
+                )
+            elseif option[2].heat_pump_configuration == "WWHP"
+                add_to_expression!(m[:InitialCapexNoIncentives], 
+                    (option[2].wwhp_heating_pump_installed_cost_curve[2]*option[2].wwhp_heating_pump_capacity_ton + option[2].wwhp_cooling_pump_installed_cost_curve[2]*option[2].wwhp_cooling_pump_capacity_ton)*m[Symbol("binGHP"*_n)][option[1]]
+                )
+            else
+                @warn "Unknown heat pump configuration provided, excluding GHP costs from initial capital costs."
+            end
+        end
+    end
+
+    if "ASHPSpaceHeater" in p.techs.all
+        add_to_expression!(m[:InitialCapexNoIncentives], 
+            p.s.ashp.installed_cost_per_kw * m[Symbol("dvPurchaseSize"*_n)]["ASHPSpaceHeater"]
+        )
+    end
+
+    if "ASHPWaterHeater" in p.techs.all
+        add_to_expression!(m[:InitialCapexNoIncentives], 
+            p.s.ashp_wh.installed_cost_per_kw * m[Symbol("dvPurchaseSize"*_n)]["ASHPWaterHeater"]
+        )
+    end
```

Comment:
I checked and we don't leave any technologies out, but we may want to note in the Wiki that if we add a technology to REopt that this function should be updated in addition to the other parts of the code that require it.

Reply:
good idea, I added this to https://github.com/NREL/REopt.jl/wiki/Steps-to-register-a-new-version-of-the-REopt-package

Reply:
looks great, thanks for doing that!